main.cpp

```diff
@@ -16,5 +18,20 @@ int main() {
   std::sort(zip.begin(), zip.end());
 
   for (const auto & z: zip) std::cout << z << std::endl;
+
+
+  std::vector<int> ai(1000000);
```

Comment:
Are you checking if it performs well? There should be very little overhead :)

Should we keep these huge vectors into the example?

Reply:
I hadn't, but I've added it now.  The main executable takes an optional length argument, defaulting to 10,000.

```
$ ./main         
[ 3, Alice ]
[ 1, Bob ]
[ 4, Charles ]
[ 2, David ]

[ 1, Bob ]
[ 2, David ]
[ 3, Alice ]
[ 4, Charles ]
Took 3.329ms to sort iterator pairs
Took 2.419ms to sort pairs
```
With 10,000,000:

```
$ ./main 10000000
[ 3, Alice ]
[ 1, Bob ]
[ 4, Charles ]
[ 2, David ]

[ 1, Bob ]
[ 2, David ]
[ 3, Alice ]
[ 4, Charles ]
Took 6751.88ms to sort iterator pairs
Took 5916.86ms to sort pairs
```
And for 100,000,000
```
$ ./main 100000000
[ 3, Alice ]
[ 1, Bob ]
[ 4, Charles ]
[ 2, David ]

[ 1, Bob ]
[ 2, David ]
[ 3, Alice ]
[ 4, Charles ]
Took 79021.7ms to sort iterator pairs
Took 72862.6ms to sort pairs
```

So there's a relatively small overhead (14-33%) for performing the sort as structs of arrays rather than arrays of structs, but considering the cost of conversion, it should be a net performance gain.